client/Annotations/index.js

```diff
@@ -126,6 +128,8 @@ Template.Annotations.events({
                 } else {
                     AnnotationFiles.insert({"filename": fileName, "info": info, "annotations": csvAnnotations});
```

Comment:
I would put this insert call in a try-catch block as well
```suggestion
	try{
                    AnnotationFiles.insert({"filename": fileName, "info": info, "annotations": csvAnnotations});
                    }
        } catch (err){
                console.log(err);
                filesUploadFailed += fileName + ": " + err + "\n";
                uploadsEnded++;
                window.prompt("There has been an error, please try again");
            }
```